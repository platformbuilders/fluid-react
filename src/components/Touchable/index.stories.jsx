import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import markdown from './touchable.md';
import Touchable from './index';

storiesOf('Touchable', module)
  .addParameters({
    readme: {
      content: markdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => (
    <Touchable onPress={action('clicked')}>Tap Here</Touchable>
  ));
