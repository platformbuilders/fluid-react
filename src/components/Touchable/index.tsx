import React from 'react';
import { TouchableOpacity } from 'react-native';
import { HapticFeedbackType } from 'react-native-haptic';
import { generateHaptic } from '../../utils/helpers';

type Props = {
  onPress: (x: any) => void;
  children: JSX.Element;
  accessibility: string;
  haptic?: HapticFeedbackType;
  touchable?: boolean;
};

const CommonTouchable: React.FC<Props> = ({
  onPress = (): void => {},
  haptic = 'impact',
  touchable = true,
  accessibility,
  ...rest
}) => (
  <TouchableOpacity
    {...rest}
    accessibilityLabel={accessibility}
    disabled={touchable}
    onPress={(e): void => {
      generateHaptic(haptic);
      onPress(e);
    }}
  />
);

export default CommonTouchable;
