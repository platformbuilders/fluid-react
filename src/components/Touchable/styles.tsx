import styled from 'styled-components';
import ButtonBase from '@material-ui/core/ButtonBase';

export const CustomTouchable = styled(ButtonBase)``;
