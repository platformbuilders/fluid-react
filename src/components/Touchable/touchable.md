
# Touchable 

<br/>
<!-- STORY -->

<br/>

### Source Code
<!-- SOURCE -->

<!-- PROPS -->


