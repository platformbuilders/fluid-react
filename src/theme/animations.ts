import LoadingAnimation from '../assets/animations/loading.json';
import { Animations } from '../types';

export default {
  circularLoading: LoadingAnimation,
  buttonLoading: LoadingAnimation,
  linearLoading: LoadingAnimation,
} as Animations;
