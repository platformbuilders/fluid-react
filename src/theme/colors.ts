import { lighten, darken } from 'polished';
import { Colors } from '../utils/types';

const primary = '#C6B09E';
const secondary = '#805281';
const tertiary = '#2A2E39';
const accent = '#547E6A';

const colors: Colors = {
  failure: '#cc0000',
  warning: '#F5B800',
  success: '#357a38',
  transparent: 'transparent',
  disabled: '#EBEBE4',
  primary: {
    light: lighten(0.05, primary),
    main: primary,
    dark: darken(0.12, primary),
    contrast: '#ffffff',
  },
  secondary: {
    light: lighten(0.05, secondary),
    main: secondary,
    dark: darken(0.1, secondary),
    contrast: '#ffffff',
  },
  tertiary: {
    light: lighten(0.15, tertiary),
    main: tertiary,
    dark: darken(0.1, tertiary),
    contrast: '#ffffff',
  },
  accent: {
    light: lighten(0.05, accent),
    main: accent,
    dark: darken(0.05, accent),
    contrast: '#ffffff',
  },
};

export default colors;
