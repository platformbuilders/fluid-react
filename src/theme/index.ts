export { default as colors } from './colors';
export { default as radius } from './radius';
export { default as animations } from './animations';
