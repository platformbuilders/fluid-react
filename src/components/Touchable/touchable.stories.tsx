import { expect, jest } from '@storybook/jest';
import type { Meta, StoryObj } from '@storybook/react';
import { configure, userEvent, within } from '@storybook/testing-library';

import Touchable from './index';

configure({ testIdAttribute: 'testid' });

//  Mocks
const mockTextId = 'touchable-test-id';
const mockDisabledTextId = 'touchable-disabled-test-id';
const mockOnPress = jest.fn();

const meta: Meta<typeof Touchable> = {
  title: 'Components/Touchable',
  component: Touchable,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  args: {
    testID: mockTextId,
    children: 'Touchable Test',
    onPress: mockOnPress,
    disabled: false,
  },
};

type Story = StoryObj<typeof Touchable>;

export const Default: Story = {
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Touchable | Test Render', async () => {
      expect(canvas.getByTestId(mockTextId)).toBeInTheDocument();
    });

    await step('Touchable | Event Click', async () => {
      expect(canvas.getByTestId(mockTextId)).not.toHaveAttribute('disabled');
      await userEvent.click(canvas.getByTestId(mockTextId));
      expect(mockOnPress).toHaveBeenCalled();
    });
  },
};

export const Disabled: Story = {
  args: {
    disabled: true,
    testID: mockDisabledTextId,
  },
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Touchable | Test Render', async () => {
      expect(canvas.getByTestId(mockDisabledTextId)).toBeInTheDocument();
    });

    await step('Touchable | Event Click', async () => {
      await userEvent.click(canvas.getByTestId(mockDisabledTextId));
      expect(canvas.getByTestId(mockDisabledTextId)).toHaveAttribute(
        'disabled',
      );
    });
  },
};

export default meta;
