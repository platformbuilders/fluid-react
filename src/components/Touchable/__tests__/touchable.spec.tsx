import { ThemeProvider } from 'styled-components';
import { faker } from '@faker-js/faker';
import { fireEvent, render } from '@testing-library/react';

import Touchable, { Props } from '..';
import theme from '../../../theme';

const defaultContent = 'Text';
const defaultProps: Props = {
  accessibility: 'accessibility',
  onPress: jest.fn(),
};

describe('Component: Touchable', () => {
  test('snapshots', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Touchable {...defaultProps}>{defaultContent}</Touchable>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  test('check props', () => {
    // when
    const mockContent = faker.random.word();
    const { getByText } = render(
      <Touchable {...defaultProps}>{mockContent}</Touchable>,
    );

    // then
    expect(getByText(mockContent)).toBeInTheDocument();
  });

  test('should call onPress when clicked', () => {
    // should
    const mockFunction = jest.fn();
    const { getByRole } = render(
      <Touchable {...defaultProps} onPress={mockFunction}>
        {defaultContent}
      </Touchable>,
    );

    // when
    expect(mockFunction).not.toHaveBeenCalled();
    fireEvent.click(getByRole('button'));

    // then
    expect(mockFunction).toHaveBeenCalled();
  });
});
