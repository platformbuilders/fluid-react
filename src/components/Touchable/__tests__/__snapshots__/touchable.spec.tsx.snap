// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Touchable snapshots 1`] = `
<div>
  <button
    accessibility="accessibility"
    class="MuiButtonBase-root sc-bczRLJ"
    tabindex="0"
    type="button"
  >
    Text
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
